File: Python/Data Structures & Algorithms Module - Python/Revision/LinkedList/LinkedListApp.py
"""
LinkedListApp.py
Billy Cussen
05/01/2021
"""

from LinkedList import LinkedList

myLinkedList = LinkedList()

print("Is Empty: "+str(myLinkedList.isEmpty()))
try:
    print("Size: "+myLinkedList.size())
except Exception as e:
    print(e)
try:
    print(myLinkedList.toString())
except Exception as e:
    print(e)


myLinkedList.add(2)
myLinkedList.add(4)
myLinkedList.add(6)
myLinkedList.addByIndex(20,0)
myLinkedList.addByIndex(40,2)
myLinkedList.addByIndex(60,myLinkedList.size())
print(myLinkedList.toString())

myLinkedList.removeByIndex(0)
myLinkedList.removeByIndex(myLinkedList.size()-1)
myLinkedList.removeByIndex(1)

print(myLinkedList.toString())

File: Java/Data Structures & Algorithms Module - Java/Revision/LinkedList/LinkedListApp.java
/*
LinkedListApp.java
Billy Cussen
05/01/2021
*/
public class LinkedListApp {
  
    public static void main(String [] args){

        LinkedList<Integer> myList = new LinkedList<Integer>();

        System.out.println("Is Empty: "+myList.isEmpty());
        try{
            System.out.println(myList.size());
        }catch(Exception e){
            System.out.println(e.toString());
        }
        try{
            System.out.println(myList.toString());
        }catch(Exception e){
            System.out.println(e.toString());
        }

        myList.add(2);
        myList.add(4);
        myList.add(6);

        System.out.println("Is Empty: "+myList.isEmpty());
        System.out.println(myList.size());
        System.out.println(myList.toString());

        myList.addByIndex(20, 0);
        myList.addByIndex(40, 2);
        myList.addByIndex(60, myList.size());

        System.out.println("Is Empty: "+myList.isEmpty());
        System.out.println(myList.size());
        System.out.println(myList.toString());

        myList.removeByIndex(0);
        myList.removeByIndex(1);
        myList.removeByIndex(myList.size()-1);

        System.out.println("Is Empty: "+myList.isEmpty());
        System.out.println(myList.size());
        System.out.println(myList.toString());
    }
}
